lib/bootloader_setup.pm

```diff
@@ -249,7 +250,24 @@ sub boot_local_disk {
         if (match_has_tag 'inst-slof') {
             diag 'specifying local disk for boot from slof';
             my $slof = 5;
-            $slof += 1 if get_var('VIRTIO_CONSOLE');
+            if (get_var('VIRTIO_CONSOLE') && is_ppc64le) {
+                type_string_very_slow "devalias";
+                send_key 'ret';
+                save_screenshot;
+                my $file = './serial0';
+                open (FILE, $file);
+                my @content = <FILE>;
+                close FILE;
+                my $out = 0;
+                foreach my $line (@content) {
+                     if ($line =~ /^disk/) {
+                         $line =~ s/\0|\r|\n//g;
+                         my @num = split(/scsi@/, $line);
+                         $out = $num[1];
```

Comment:
maybe we need break loop if we get the num

Reply:
Agree, besides in my opinion to find the id in so long a file as serial0 will need long time, so I think it is a successful solution while maybe not the best solution. :)